src/Service.Tests/SqlTests/RestApiTests/Insert/MsSqlInsertApiTests.cs

```diff
@@ -378,6 +390,65 @@ public async Task InsertOneInViewBadRequestTest()
             );
         }
 
+        /// <summary>
+        /// Test to validate that even when an insert DML trigger is enabled on a table, we still return the
+        /// latest data as it is present after the trigger gets executed. Whenever an insert DML trigger is enabled,
+        /// we use a subsequent SELECT query to get the data instead of using OUTPUT clause. To validate that the data is returned
+        /// as it is after the trigger is executed, we use the values which are updated by the trigger in the WHERE predicates of the verifying sql query.
+        /// </summary>
+        [TestMethod]
+        public async Task InsertOneInTableWithInsertTrigger()
+        {
+            // Validate that insert operation succeeds when an insert DML trigger is enabled for a table
+            // with autogenerated primary key. For table containing an autogenerated PK field (id), and non-autogenerated PK field (u_id),
+            // the values of the non-autogenerated PK fields are inserted into a temporary table and then an INNER JOIN is performed
+            // based on those fields' values from the actual table and the temporary table. For the autogenerated PK field, an additional
+            // WHERE predicate is added to fetch the unique record by extracting the value of the autogenerated PK field using the
+            // SCOPE_IDENTITY() method provided by Sql Server.
+            // Trigger behavior: The trigger updates the salary to max(0,min(100,salary)).
```

Comment:
Refer to comment https://github.com/Azure/data-api-builder/pull/1630#discussion_r1325210147